File: androidApp/src/main/kotlin/aveek/portfolio/mobile/marketpulse/presentation/util/StockFormatter.kt
package aveek.portfolio.mobile.marketpulse.presentation.util

import java.util.Locale
import kotlin.math.abs
import kotlin.math.roundToInt

fun formatPrice(price: Double): String {
    val rounded = (price * 100).roundToInt() / 100.0
    return rounded.toString()
}

fun formatChange(change: Double): String {
    val rounded = (change * 100).roundToInt() / 100.0
    return if (rounded > 0) "+$rounded" else rounded.toString()
}

fun formatChangePercent(changePercent: String): String {
    val normalized = changePercent.trim().removeSuffix("%")
    val value = normalized.toDoubleOrNull() ?: return changePercent.ifBlank { "-" }
    val magnitude = String.format(Locale.US, "%.2f", abs(value))

    return when {
        value > 0 -> "+$magnitude%"
        value < 0 -> "-$magnitude%"
        else -> "0.00%"
    }
}

File: iosApp/iosApp/Presentation/Util/StockFormatter.swift
import Foundation

enum StockFormatter {
    static func formatPrice(_ price: Double) -> String {
        let rounded = (price * 100).rounded() / 100
        return String(rounded)
    }

    static func formatChange(_ change: Double) -> String {
        let rounded = (change * 100).rounded() / 100
        return rounded > 0 ? "+\(rounded)" : String(rounded)
    }

    static func formatChangePercent(_ changePercent: String) -> String {
        var normalized = changePercent.trimmingCharacters(in: .whitespaces)
        if normalized.hasSuffix("%") {
            normalized = String(normalized.dropLast())
        }
        guard let value = Double(normalized) else {
            return changePercent.trimmingCharacters(in: .whitespaces).isEmpty ? "-" : changePercent
        }
        let magnitude = String(format: "%.2f", abs(value))
        switch value {
        case let v where v > 0: return "+\(magnitude)%"
        case let v where v < 0: return "-\(magnitude)%"
        default: return "0.00%"
        }
    }
}
